ref/general/hardening-intro.adoc

```diff
@@ -22,49 +22,38 @@ Hardening is a general term that is used to describe the measures that are taken
+To learn more about network vulnerabilities and ways to prevent network intrusions, go to <link pending>.
 
-===== Use Transport Security
-It's important to use the feature:transportSecurity-1.0[] feature to ___. This feature enables the feature:ssl-1.0[] feature In addition to enabling `transportSecurity-1.0`, disable non-SSL ports by setting them to `-1` in the `server.xml` file. Only use certificates that are signed by a Certificate Authority for public-facing production servers.
+==== Best practices
 
-===== alsdkfj;asdf
-Configure applications to run on a virtual host to keep admin traffic separate from application requests.
+* *Use a firewall*: Use a firewall to shield production Open Liberty servers so that they don't run in the DMZ. The DMZ is an unsecured network segment that exists between internal and external networks. Firewalls help prevent security intrusions by monitoring network traffic and blocking specific traffic based on security rules.
+* *Enable the `transportSecurity` feature*: The feature:transportSecurity-1.0[] feature enables the feature:ssl-1.0[] feature to support SSL connections and adds the ability to configure outbound transports. In addition to enabling `transportSecurity-1.0`, disable non-SSL ports by setting them to `-1` in the `server.xml` file, and use only certificates that are signed by a certificate authority for public-facing production servers.
+* *Run applications on a virtual host*: Configure applications to run on a virtual host. Open Liberty admin traffic comes in on the default host, so configuring your applications to run on a virtual host keeps Open Liberty admin traffic separate from application requests.
 
 
 [#application-configuration]
 == Application configuration
 Application configuration intrusions occur when external users run applications that derive or inherit privileges that they are not authorized to. One example of this type of intrusion happens when an application inherits the identity of the server, giving the application unauthorized permissions.
 
-To learn more about application configuration vulnerabilities and ways to prevent application intrusions, go to <link to application configuration topic>.
+To learn more about application configuration vulnerabilities and ways to prevent application intrusions, go to <link pending>.
 
-=== Best practices
+==== Best practices
 
-===== alskdjfa;
+* *Understand roles and priorities*: Roles and mappings are crucial for securely configuring your application. Incorrect role mapping can give inadvertent levels of access to users, which can result in some users with access to sensitive resources. You define roles by using annotations and the `web.xml` file, and you bind these roles to users or groups in the `ibm-application-bnd.xm` and `server.xml` files. For more information about roles and mapping them to the appropriate users or groups, go to <link pending>.
```

Comment:
`You define roles by using annotations and the `web.xml` file, and you bind these roles to users or groups in the `ibm-application-bnd.xm` and `server.xml` files.`

This is just one way to specify and define roles- since you're not actually telling how to do this, might be better not to get into it here and just send them to the Authorization topic. I think you could omit this sentence (unless it was specifically requested) and the section would still work the same.

Reply:
Updated